components/merkledb/src/views/metadata.rs

```diff
@@ -339,11 +375,78 @@ impl<T: RawAccess> IndexesPool<T> {
     }
 }
 
+impl<T: RawAccessMut> IndexesPool<T> {
+    /// Moves indexes with the specified prefix from the next version (i.e., `^prefix.*` form)
+    /// to the current version (`prefix.*` form). The existing old indexes are replaced, or
+    /// removed if the new index is a `Tombstone`. If there is no overriding index, an old
+    /// index is left in place.
+    ///
+    /// # Return value
+    ///
+    /// Returns resolved addresses of the removed indexes. For each address, we also return a flag
+    /// indicating whether the corresponding index was removed from aggregation (i.e., was
+    /// aggregated and was not replaced by an aggregated index).
+    pub(crate) fn flush_migration(&mut self, prefix: &str) -> Vec<(ResolvedAddress, bool)> {
+        let prefix = format!("^{}.", prefix);
```

Comment:
For hot code paths it is reasonable to use `[a, b, c].concat()` instead of `format!` macro.